sklearn/metrics/cluster/tests/test_common.py

```diff
@@ -0,0 +1,193 @@
+METRICS_NORMALIZED_OUTPUT = [
+    "adjusted_rand_score", "homogeneity_score", "completeness_score",
+    "v_measure_score", "adjusted_mutual_info_score", "fowlkes_mallows_score",
+    "normalized_mutual_info_score"
+]
+
+
+def assert_between(var, score_1, score_2):
+    """ Returns a boolean value
+    Helper function to check if score lies in between two values
+    """
+    if assert_greater(var, score_1) and assert_less(var, score_2):
+        return True
+    else:
+        return False
+
+
+def test_symmetry():
+    rng = np.random.RandomState(0)
+    y1 = rng.randint(3, size=30)
+    y2 = rng.randint(3, size=30)
+    for name in SYMMETRIC_METRICS:
+        metric = SUPERVISED_METRICS_DICT[name]
+        assert_almost_equal(metric(y1, y2),
+                            metric(y2, y1))
+
+    for name in NON_SYMMETRIC_METRICS:
+        metric = SUPERVISED_METRICS_DICT[name]
+        assert_almost_equal(metric([0, 1, 2, 5, 4, 9], [0, 1, 9, 4, 3, 5]),
+                            metric([0, 1, 9, 4, 3, 5], [0, 1, 2, 5, 4, 9]))
+
+
+def test_exactly_zero_info_score():
+    # Check numerical stability when information is exactly zero
+    for i in np.logspace(1, 4, 4).astype(np.int):
+        labels_a, labels_b = (np.ones(i, dtype=np.int),
+                              np.arange(i, dtype=np.int))
+        for name in METRICS_ZERO_INFO:
+            metric = SUPERVISED_METRICS_DICT[name]
+            assert_almost_equal(metric(labels_a, labels_b), 0.0)
+
+
+def test_normalized_output():
+    upper_bound_1 = [0, 0, 0, 1, 1, 1]
+    upper_bound_2 = [0, 0, 0, 1, 1, 1]
+    for name in METRICS_NORMALIZED_OUTPUT:
+        metric = SUPERVISED_METRICS_DICT[name]
+        assert_between(metric([0, 0, 0, 1, 1], [0, 0, 0, 1, 2]), 0.0, 1.0)
+        assert_between(metric([0, 0, 1, 1, 2], [0, 0, 1, 1, 1]), 0.0, 1.0)
+        assert_equal(metric(upper_bound_1, upper_bound_2), 1.0)
+
+    # For symmetric metrics the lower bound is defined
+    lower_bound_1 = [0, 0, 0, 0, 0, 0]
+    lower_bound_2 = [0, 1, 2, 3, 4, 5]
+    for name in SYMMETRIC_METRICS:
+        metric = SUPERVISED_METRICS_DICT[name]
+        assert_equal(metric(lower_bound_1, lower_bound_2), 0.0)
+
+
+# All clustering metrocs do not change score due to permutations of labels
+# that is when 0 and 1 exchchanged.
+def test_permute_labels():
+    for name in SUPERVISED_METRICS_DICT:
+        metric = SUPERVISED_METRICS_DICT[name]
+        y_label = np.array([0, 0, 0, 1, 1, 0, 1])
+        y_pred = np.array([1, 0, 1, 0, 1, 1, 0])
+        score_1 = metric(y_pred, y_label)
+        assert_equal(score_1, metric(1 - y_pred, y_label))
+        assert_equal(score_1, metric(1 - y_pred, 1 - y_label))
+        assert_almost_equal(score_1, metric(y_pred, 1 - y_label))
+
+    # Test for Silhouette_score
+    dataset = datasets.load_iris()
+    X = dataset.data
+    y_pred = dataset.target
+    D = pairwise_distances(X, metric='euclidean')
+    score_1 = silhouette_score(D, y_pred, metric='precomputed')

```

Comment:
I don't think we can expect to handle precomputed here. Each metric's unit tests can handle such quirks.